prive/datasets/data_description.py

```diff
@@ -0,0 +1,80 @@
+
+
+class DataDescription:
+    def __init__(self, description):
+        self.schema = description
+
+    @property
+    def num_features(self):
+        """
+        int : Number of columns.
+
+        """
+        return len(self.schema)
+
+    @property
+    def encoded_dim(self):
+        """
+        int : Number of dimensions the data would have if encoded. This assumes
+        ordered and infinite variables will have one dimension, and only finite,
+        unordered variables would be one-hot encoded, where they will require
+        one dimension per category.
+
+        """
+        nfeatures = 0
+
+        for cdict in self:
```

Comment:
Slightly confused by python here. Why isn't this `for cdict in self.schema`? 

Also ... `schema` will be a list, not a dict (if indeed that's what was meant here, but as noted I am confused.)

Reply:
Okay, going to leave this in for now, make the change above, and merge.

Reply:
So the implementation of `__getitem__` below essentially allows us to iterate over `self.schema` by just calling iteration over `self`. This class is essentially a list with some extra methods and some alternative indexing (can index by column name). This saves us from having to always write `description.schema`.